onyo/lib/commands.py

```diff
@@ -513,22 +513,39 @@ def onyo_get(inventory: Inventory,
         raise ValueError(f"Allowed sorting modes: {', '.join(allowed_sorting)}")
 
     selected_keys = selected_keys or inventory.repo.get_asset_name_keys() + ['path']
-    results = inventory.get_assets_by_query(include=include,
-                                            exclude=exclude,
-                                            depth=depth,
-                                            match=match)
-    results = list(fill_unset(results, selected_keys))
-    # convert paths for output
+    results = list(inventory.get_assets_by_query(include=include,
+                                                 exclude=exclude,
+                                                 depth=depth,
+                                                 # pyre's complaint boils down to "Dict" not being "Dict | UserDict".
+                                                 # This is useless nonsense.
+                                                 match=match))  # pyre-ignore[6]
+
+    # convert paths for sorting and printing
     for r in results:
         r['path'] = r['path'].relative_to(inventory.root)
 
+    # Note: Sorting is done before any further filtering/replacing. Therefore, one can sort by keys that aren't actually
+    #       in the output of the command. This behavior is utilized in tests.
     results = natural_sort(
         assets=results,
         # pyre can't tell SORT_ASCENDING is not an arbitrary string but matches the Literal declaration:
         keys=sort or {'path': SORT_ASCENDING})  # pyre-ignore[6]
 
-    # filter output for `keys` only
-    results = [{k: v for k, v in r.items() if k in selected_keys} for r in results]
+    # Filter results for `selected_keys` first in order to not iterate over irrelevant parts of the assets in subsequent
+    # filters, replacements, etc.
+    results = [{k: r[k] if k in r else UNSET_VALUE
+                for k in selected_keys}
+               for r in results]
+    # Note: We now have a list of regular dicts. DotNotation would be resolved at this point by having keys with
+    #       literal dots.
+
+    # Replace structures by an indication of type.
+    # Instead of outputting `{}`, `[]` or even `{some: {other: value}}`, just print `<dict>`/`<list>`.
+    # If information on content was wanted, the respective `keys` should be specified instead.
```

Comment:
Nit: "was" -> "is" (only commenting because the confusion from the above "would" has continued to prime me. ;-)